src/music/dmusic.cpp

```diff
@@ -429,14 +429,13 @@ bool DLSFile::ReadDLSWaveList(FILE *f, DWORD list_length)
 	return true;
 }
 
-bool DLSFile::LoadFile(const wchar_t *file)
+bool DLSFile::LoadFile(const std::filesystem::path &file)
 {
 	Debug(driver, 2, "DMusic: Try to load DLS file {}", FS2OTTD(file));
 
-	FILE *f = _wfopen(file, L"rb");
-	if (f == nullptr) return false;
-
-	FileCloser f_scope(f);
+	auto of = FileHandle::OpenNative(file, "rb");
```

Comment:
Why is OpenNative (or _wfopen) required here? Or rather, I see the difference between the two but don't follow why this is the only needed usage of it

Reply:
Because paths have two encodings. One is our default UTF-8, and the other is operating system native which under Windows is of course `wchar_t` and UTF-16.

Since the path is gathered from the registry and uses Windows specific path functions, the calling code already has the (Windows) native path, so there's little point in converting it to `UTF-8` and then back to its native encoding.